lib/vanagon/platform/windows.rb

```diff
@@ -221,6 +222,73 @@ def output_dir(target_repo = "")
         File.join("windows", target_repo, @architecture)
       end
 
+      # Generate correctly formatted wix elements that match the
+      # structure of the directory input
+      #
+      # @param directories, Array of hashes including at least :path
+      # and optionally:
+      # @return [string] correctly formatted wix element string
+      def generate_wix_dirs(project)
+        if project.get_directories.empty?
+          raise Vanagon::Error, 'ERROR No directories specified!'
+        else
+          directories = []
+          project.get_directories.map { |dir| directories.push({ :path => dir.path, :id => dir.wix_id }) }
+
+          root = { :children => [] }
+
+          # iterate over all paths specified and break each one
+          # in to its specific directories. This will generate_wix_dirs
+          # an n-ary tree structure matching the specs from the input
+          directories.each do |dir|
+            curr = root
+            names = dir[:path].sub(/\//, '').split(/\//)
+            names.each do |name|
+              new_obj = { :name => name, :id => nil, :children => [] }
+              if (child_index = includes_child(new_obj, curr[:children]))
+                curr = curr[:children][child_index]
+              else
+                curr[:children].push(new_obj)
+                curr = new_obj
+              end
+            end
+            #at this point, curr will be the top dir
+            curr[:id] = dir[:id]
+          end
+          return generate_wix_from_graph(root)
+        end
+      end
+
+      # Find if child element is the same as one of
+      # the old_children elements, return that child
+      def includes_child(new_child, old_children)
+        old_children.each_with_index do |curr_old_child, index|
+          return index if curr_old_child[:name] == new_child[:name]
+        end unless old_children.empty?
+        return nil
+      end
+
+      # Recursively generate wix element structure
+      #
+      # @param root, the (empty) root of an n-ary tree containing the
+      # structure of directories
+      def generate_wix_from_graph(root)
+        string = ''
+        unless root[:children].empty?
```

Comment:
Why do we have to check if `root[:children]` is empty? We should just be able to loop through and return string. I guess this would fail if `root[:children]` were nil and there was no array to loop over... But on line 238 you're setting it explicitely. I feel like that's an assumption we should be safe making, in which case lines 277, 288, and 289 shouldn't be necessary.